decode.go

```diff
@@ -962,35 +969,59 @@ func unmarshalListValue(
 // Structs can represent either a message or a map.
 // For messages, the message descriptor can be provided or inferred from the node.
 // For maps, the field descriptor can be provided to validate the map keys/values.
-func unmarshalStruct(
-	unm *unmarshaler,
+func (u *unmarshaler) unmarshalStruct(
 	node *yaml.Node,
 	message *structpb.Struct,
 	msgDesc protoreflect.MessageDescriptor,
 	field protoreflect.FieldDescriptor,
+	ignoreDynamicType bool, // If true, ignore @type fields.
 ) {
 	if field != nil {
 		// Validate the field is a message or map.
 		if !field.IsMap() && field.Kind() != protoreflect.MessageKind {
```

Comment:
Admittedly unrelated, but this also needs `&& field.Kind() != protoreflect.GroupKind`.